test/function/matrix/sqrtm.test.js

```diff
@@ -11,40 +12,39 @@ describe('sqrtm', function () {
   var BB = [[ 7, 10], [15, 22]];
 
   it('should return the principal square root of a matrix', function() {
-    math.deepEqual(math.sqrtm(AA), math.matrix(A));
-    math.deepEqual(math.sqrtm(BB), math.matrix(B));
+    approx.deepEqual(math.sqrtm(AA), A);
+    approx.deepEqual(math.sqrtm(BB), B);
 
-    math.deepEqual(
+    approx.deepEqual(
       math.sqrtm(
         [[ 5, -4,  1,  0,  0],
          [-4,  6, -4,  1,  0],
          [ 1, -4,  6, -4,  1],
          [ 0,  1, -4,  6, -4],
          [ 0,  0,  1, -4,  6]]),
-      math.matrix(
         [[ 2.001450032980806, -0.997069854672086,  0.004226841220338,  0.004648098208404,  0.003162179891248],
          [-0.997069854672086,  2.006191486385745, -0.990363307725271,  0.011838782789231,  0.009369460553432],
          [ 0.004226841220338, -0.990363307725271,  2.017072849046023, -0.974622709753106,  0.026274470491696],
          [ 0.004648098208404,  0.011838782789231, -0.974622709753106,  2.050268428894304, -0.919971837350421],
-         [ 0.003162179891248,  0.009369460553432,  0.026274470491696, -0.919971837350421,  2.269992000979243]]));
+         [ 0.003162179891248,  0.009369460553432,  0.026274470491696, -0.919971837350421,  2.269992000979243]]);
   });
 
   it('should return the principal square root of a matrix with just one value', function() {
-    math.deepEqual(math.sqrtm([ 4]), [2]);
-    math.deepEqual(math.sqrtm([16]), [4]);
-    math.deepEqual(math.sqrtm([20.25]), [4.5]);
+    assert.deepEqual(math.sqrtm([ 4]), [2]);
+    assert.deepEqual(math.sqrtm([16]), [4]);
+    assert.deepEqual(math.sqrtm([20.25]), [4.5]);
   });
 
   it('should return the principal square root of a matrix of big numbers', function() {
-    math.deepEqual(math.sqrtm(math.bignumber(AA)), math.matrix(math.bignumber(A)));
-    math.deepEqual(math.sqrtm(math.bignumber(BB)), math.matrix(math.bignumber(B)));
+    // approx.deepEqual(math.sqrtm(math.bignumber(AA)), math.bignumber(A));
+    // approx.deepEqual(math.sqrtm(math.bignumber(BB)), math.bignumber(B));
```

Comment:
Why is `approx.deepEqual()` not working as expected for *Matrices* of `BigNumbers`?

Reply:
Hmm... But this goes against the ideology of having input `Array` → `Array` output, and `Matrix` → `Matrix`, no? Both the method I implemented in this PR and the already existing methods should be sound concerning whatever the contents of an `Array` or `Matrix` are, whether they are regular floats or something else (such as a `BigNumber`).

Reply:
Ah sorry, I read your changes wrongly, your commented lines should indeed work.

When I do some logging, I see that the output contains round off errors, i.e. for BigNumbers the solution does not fully converge: 

```js
console.log(math.format(math.sqrtm(math.bignumber(AA))))
// [
//   [
//     5.000000000000000000000000931510855380425361089387837696247148415,
//     2.000000000000000000000000931510855380425361089387837696247148416
//   ], 
//   [
//     4.000000000000000000000001863021710760850722178775675392494296833, 
//     7.00000000000000000000000186302171076085072217877567539249429683
//   ]
// ]
```

So we can either find a different example that does converge, or first round the result to say 12 digits using `math.round` and then compare with the expected result.